### PR TITLE
JAMES-4077 [SearchSnippet] Highlight OpenSearch implementation

### PR DESCRIPTION
**Pull request by vttranlina**
jira https://issues.apache.org/jira/browse/JAMES-4077
resolve https://github.com/linagora/james-project/issues/5253

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/searchhighligt/SearchHighLighterContract.java [hunk 1]
```diff
@@ -64,14 +65,14 @@ default void highlightSearchShouldReturnHighLightedSubjectWhenMatched() throws E
         ComposedMessageId m1 = appendMessage(MessageManager.AppendCommand.from(
                 Message.Builder.of()
                     .setTo("to@james.local")
-                    .setSubject("Hallo! Thx Matthieu for your help")
+                    .setSubject("Hallo, Thx Matthieu for your help")
```
Comment (vttranlina) [inline mailbox/api/src/test/java/org/apache/james/mailbox/searchhighligt/SearchHighLighterContract.java:68]: I don't understand why the character "!" causes the highlight results to exclude `Hallo!`. 

I attempted various highlight configurations while querying, but none worked as expected. When indexing `subject`, I replaced the tokenizer `uax_url_email` with `standard`, but that didn't work either. 

To simplify the test case for Lucene and OpenSearch, I changed `!` to `,`.

File: mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/search/OpenSearchSearchHighlighter.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            .map(jsonData -> jsonData.toJson().asJsonArray().getString(0))
+            .map(messageIdFactory::fromString)
+            .orElseThrow(() -> new IllegalStateException("Can not extract MessageID for search result: " + searchResult.id()));
+
+        Map<String, List<String>> highlightHit = searchResult.highlight();
+
+        Optional<String> highlightedSubject =  Optional.ofNullable(highlightHit.get(JsonMessageConstants.SUBJECT))
+            .map(List::getFirst);
+        Optional<String> highlightedTextBody = Optional.ofNullable(highlightHit.get(JsonMessageConstants.TEXT_BODY))
+            .map(List::getFirst);
```
Comment (chibenwa) [inline mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/search/OpenSearchSearchHighlighter.java:81]: While we are at it any chance we also highlight within attachment text content?

**Review by chibenwa — COMMENTED** [review]
Look cool indeed!

**Review by Arsnael — COMMENTED** [review]
Read it, looks good, nothing to add

**Review by quantranhong1999 — COMMENTED** [review]
Looks good to me.

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**